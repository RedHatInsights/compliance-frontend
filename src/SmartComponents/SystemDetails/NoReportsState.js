import React from 'react';
import propTypes from 'prop-types';
import { CloudSecurityIcon } from '@patternfly/react-icons';
import {
  Title,
  Bullseye,
  EmptyState,
  EmptyStateBody,
  EmptyStateIcon,
} from '@patternfly/react-core';

const NoReportsState = ({ system }) =>
  system?.hasPolicy && !system?.testResultProfiles?.length ? (
    <Bullseye>
      <EmptyState>
        <EmptyStateIcon
          icon={CloudSecurityIcon}
          title="Compliance"
          size="xl"
          style={{
            fontWeight: '500',
            color: 'var(--pf-global--primary-color--100)',
          }}
        />
        <Title headingLevel="h1" size="lg">
          No results reported
        </Title>
        <EmptyStateBody>
          This system is part of {system?.policies?.length}
          {system?.policies?.length > 1 ? ' policies' : ' policy'}, but has not
          returned any results.
        </EmptyStateBody>
        <EmptyStateBody>
          Reports are returned when the system checks into Insights. By default,
          systems check in every 24 hours.
        </EmptyStateBody>
      </EmptyState>
    </Bullseye>
  ) : (
    <></>
  );

NoReportsState.propTypes = {
  system: propTypes.shape({
    hasPolicy: propTypes.bool,
    testResultProfiles: propTypes.array,
    policies: propTypes.array,
  }),
};

export default NoReportsState;
