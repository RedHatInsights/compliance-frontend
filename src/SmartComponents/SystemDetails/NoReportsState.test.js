import NoReportsState from './NoReportsState';

describe('NoReportsState', () => {
  it('with a system having multiple policies', () => {
    const wrapper = shallow(
      <NoReportsState system={{ hasPolicy: true, policies: [{}, {}] }} />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('with a system having policies', () => {
    const wrapper = shallow(
      <NoReportsState system={{ hasPolicy: true, policies: [{}] }} />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('with a system having no policies', () => {
    const wrapper = shallow(
      <NoReportsState system={{ hasPolicy: false, policies: [] }} />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('with a system missing policies prop', () => {
    const wrapper = shallow(<NoReportsState system={{ hasPolicy: false }} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('with a system having missing hasPolicy prop', () => {
    const wrapper = shallow(<NoReportsState system={{}} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('with a system having a policy but missing test results', () => {
    const wrapper = shallow(
      <NoReportsState system={{ testResultProfiles: [] }} />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('with a system having a policy and test results', () => {
    const wrapper = shallow(
      <NoReportsState system={{ testResultProfiles: [{}] }} />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
