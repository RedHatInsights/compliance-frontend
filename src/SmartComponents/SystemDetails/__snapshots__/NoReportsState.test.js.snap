// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoReportsState with a system having a policy and test results 1`] = `<Fragment />`;

exports[`NoReportsState with a system having a policy but missing test results 1`] = `<Fragment />`;

exports[`NoReportsState with a system having missing hasPolicy prop 1`] = `<Fragment />`;

exports[`NoReportsState with a system having multiple policies 1`] = `
<Bullseye>
  <EmptyState>
    <EmptyStateIcon
      icon={[Function]}
      size="xl"
      style={
        Object {
          "color": "var(--pf-global--primary-color--100)",
          "fontWeight": "500",
        }
      }
      title="Compliance"
    />
    <Title
      headingLevel="h1"
      size="lg"
    >
      No results reported
    </Title>
    <EmptyStateBody>
      This system is part of 
      2
       policies
      , but has not returned any results.
    </EmptyStateBody>
    <EmptyStateBody>
      Reports are returned when the system checks into Insights. By default, systems check in every 24 hours.
    </EmptyStateBody>
  </EmptyState>
</Bullseye>
`;

exports[`NoReportsState with a system having no policies 1`] = `<Fragment />`;

exports[`NoReportsState with a system having policies 1`] = `
<Bullseye>
  <EmptyState>
    <EmptyStateIcon
      icon={[Function]}
      size="xl"
      style={
        Object {
          "color": "var(--pf-global--primary-color--100)",
          "fontWeight": "500",
        }
      }
      title="Compliance"
    />
    <Title
      headingLevel="h1"
      size="lg"
    >
      No results reported
    </Title>
    <EmptyStateBody>
      This system is part of 
      1
       policy
      , but has not returned any results.
    </EmptyStateBody>
    <EmptyStateBody>
      Reports are returned when the system checks into Insights. By default, systems check in every 24 hours.
    </EmptyStateBody>
  </EmptyState>
</Bullseye>
`;

exports[`NoReportsState with a system missing policies prop 1`] = `<Fragment />`;
